### DISCUSSION TITLE
Discussing Bluetooth support

### DISCUSSION THREAD

**Discussion by ghost** [localsend/localsend#427] [Discussion (English, International)]
When there is no Wifi, Bluetooth could be the only way to communicate locally, and I was in this situation recently. Can we discuss potential Bluetooth support here and consider if it is a viable feature?

**Comment by rpc31** [comment]
I've been exactly in the same situation. Trying to share files to people in the same car and train. Any idea how to implement that?

**Comment by logaway** [comment]
It's not a good idea to transfer files via Bluetooth. Instead, you can enable the hotspot service to connect all the devices to the same Wi-Fi network.

**Reply by rpc31** [reply]
I’ve never heard of such things, especially for occasionally small file transferts (typically some documents, pictures and videos) in a non-professional setup. 

Where, in IOS, is that service you’re referring to ?

**Comment by walzph** [comment]
This feature is exactly what distinguishes LocalSend from AirDrop. 

AirDrop does not rely on some local network, but uses Bluetooth LE for device discovery and subsequently creates a temporary WiFi Direct connection for Peer-to-Peer transfer. 

Using only Bluetooth for file transfers is indeed not optimal, due to limited transfer speed, range constraints, higher interference and some security considerations. 

The combination of both (Bluetooth LE and Wi-Fi Direct) would be optimal, but not every wireless chip is Wi-Fi Direct compatible. 

Probably that's the reason why there is still no actual cross-platform alternative to AirDrop.

Nevertheless, LocalSend is still an awesome project. It is simple, better than relying on cloud-based solutions and just works in most situations.

Looking forward to your opinions on implementing BLE and Wi-Fi Direct :)

**Reply by Tienisto** [reply]
Unfortunately, iOS does not support wifi direct: https://developer.apple.com/forums/thread/24661

**Reply by Bronts** [reply]
In my company, desktop office computers are only connected to the corporate Intranet, and wireless cards are not allowed, but separate laptops are allowed to access the Internet to look up some documents. Sometimes it is necessary to transfer some documents under a few hundred KB between the laptop and the desktop computer, and the use of the USB disk is more troublesome. So using Bluetooth should be more convenient. In addition, Windows built-in Bluetooth transfer can be used for file transfer, but the operation is cumbersome. It is hoped that the LocalSend team will fully consider such use cases.

**Comment by talperets** [comment]
I don't understand why it can't be another option on top of sending files over a local network.

**Reply by Tienisto** [reply]
As I've said in another comment: the bluetooth API is not easy to work with. Especially not on iOS. If you have a prototype, feel free to share it.

**Reply by JaredTweed** [reply]
Would you accept a pull request that would make it work for android/linux/windows but not iOS? My thought is that it would be designed so that if a device is already on the same network, it would use that, but if not, then it would transfer via bluetooth.

**Comment by WillemJeo** [comment]
Can WebRTC DataChannel be used as an alternative for discovering devices and transferring files over the same Wi-Fi network? @Tienisto

**Reply by Tienisto** [reply]
It is planned

**Comment by WhyNotHugo** [comment]
I think what you want here is "Neighbour Awareness Networking (NAN)", also known as "Wi-Fi Aware", where two devices can find each other and set up an ephemeral Wi-Fi network for the exchange.

AirDrop is moving to use this standard protocol, so adopting it would yield great interoperability.

**Reply by KargJonas** [reply]
this would be brilliant! the usual use case for me is sending something to a friend's phone while we're on the go. currently you have to set up mobile hotspots to use localsend in this situation. would be pretty great if that was not necessary!

**Reply by richardwonka** [reply]
has there been any development on this?
